pkg/util/collections/includes_excludes.go

```diff
@@ -154,6 +155,46 @@ func ValidateIncludesExcludes(includesList, excludesList []string) []error {
 	return errs
 }
 
+// ValidateNamespaceIncludesExcludes checks provided lists of included and
+// excluded namespaces to ensure they are a valid set of IncludesExcludes data.
+func ValidateNamespaceIncludesExcludes(includesList, excludesList []string) []error {
+	errs := ValidateIncludesExcludes(includesList, excludesList)
+
+	includes := sets.NewString(includesList...)
+	excludes := sets.NewString(excludesList...)
+
+	for _, itm := range includes.List() {
+		// Although asterisks is not a valid Kubernetes namespace name, it is
+		// allowed here.
+		if itm != "*" {
+			if err := validateNamespaceName(itm); err != nil {
+				errs = append(errs, err)
+			}
+		}
+	}
+
+	for _, itm := range excludes.List() {
+		// Asterisks in excludes list have been checked previously.
+		if itm != "*" {
+			if err := validateNamespaceName(itm); err != nil {
+				errs = append(errs, err)
+			}
+		}
+	}
+
+	return errs
+}
+
+func validateNamespaceName(ns string) error {
+	if errs := validation.ValidateNamespaceName(ns, false); errs != nil {
+		for _, e := range errs {
```

Comment:
Can we return the list rather than only a single error?  Since we're appending to the list anyhow.  If there's a good reason to only return the first error, let's return errs[0] instead of the for loop.  Preferably we return the whole list though.

Reply:
Yup, will do.